lib/menu.rb

```diff
@@ -0,0 +1,22 @@
+class Menu
+  attr_reader :dishes
+
+  def initialize
+    @dishes = []
+  end
+
+  def add(dish)
+    fail "This dish is already in the menu" if contains? dish
+
+    @dishes.push(dish)
+    "#{dish.name} added to the menu"
+  end
+
+  def list
```

Comment:
This method could be named to reflect that you are formatting a string, not just returning a list.

Reply:
Also, since you are mostly formatting the dish, I'm wondering if it would be better to delegate formatting to the dish class.